clloader/datasets/core50.py

```diff
@@ -0,0 +1,79 @@
+import glob
+import os
+from typing import List, Tuple, Union
+
+import numpy as np
+
+from clloader.datasets.base import _ContinuumDataset
+
+
+class CORe50(_ContinuumDataset):
+
+    def __init__(
+        self, folder: str, train_image_ids: Union[str, List[str]], download: bool = True, **kwargs
+    ):
+        super().__init__(download=download, **kwargs)
+
+        self.folder = folder
+        self.train_image_ids = train_image_ids
+
+        if download:
+            self._download()
+
+    @property
+    def data_type(self):
+        return "image_path"
+
+    def _download(self):
+        if os.path.exists(self.folder):
+            print("CORe50 already downloaded.")
+        else:
+            raise IOError("Download it yourself.")
+
+    def init(self, train: bool) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+        x, y, t = [], [], []
+
+        train_images_ids = set()
+        if isinstance(self.train_image_ids, str):
+            with open(self.train_image_ids, "r") as f:
+                next(f)
+                for line in f:
+                    image_id = line.split(",")[0].split(".")[0]
+                    train_images_ids.add(image_id)
+        else:
+            train_images_ids = set(self.train_image_ids)
+
+        domain_counter = 0
+        for domain_id in range(10):
+            domain_folder = os.path.join(self.folder, "core50_128x128", f"s{domain_id + 1}")
+
+            has_images = False
+            for object_id in range(50):
+                object_folder = os.path.join(domain_folder, f"o{object_id + 1}")
+
+                for path in os.listdir(object_folder):
+                    image_id = path.split(".")[0]
+
+                    if train and image_id not in train_images_ids:
+                        continue
+                    elif not train and image_id in train_images_ids:
+                        continue
+
+                    x.append(os.path.join(object_folder, path))
+                    y.append(object_id)
+                    if train:
+                        t.append(domain_counter)
+                    else:
+                        t.append(0)
```

Comment:
In the NI scenario, it is possible to identify a specific test image to a specific task, would it be usefull to do it?

Reply:
Sure, but what about this case where train and test domains are different?

Train may have the domain ids [0, 1, 2, 3, 4, 5, 6]; and test [0, 1, 2]. But train#0 and test#0 are two different domains.

To avoid thinking it was the same domain, I prefered setting a dummy one for all.

Do you see a better solution?